app/controllers/static_page_controller.rb

```diff
@@ -9,16 +9,12 @@ def no_permissions
   # Gets the data of a widget
   # GET widget_data
   def widget_data
-    widget = Widget.find(params[:widget_id])
-    data = widget.get_filtered_dataset false, 10000
-    metadata = Dataset.get_metadata_for_frontend(session[:user_token], widget.dataset_id)
+    widget = WidgetService.widget(params[:widget_id])
```

Comment:
It looks like this is how we need to fetch the widgets now, also you can see that visualization is grabbed from widget_config now. 

lets make sure it works with the new stuff also

Reply:
and we dont need data, and metadata, all we need is provided from the widget config now.